tools/testrunner/testrunner.go

```diff
@@ -69,30 +70,45 @@ type attempt struct {
 	suites       junit.Testsuites
 }
 
-func (a *attempt) run(ctx context.Context, args []string) error {
+func (a *attempt) run(ctx context.Context, args []string) (error, string) {
 	args = append([]string{"--junitfile", a.junitXmlPath}, args...)
 	log.Printf("starting test attempt %d with args: %v", a.number, args)
 	cmd := exec.CommandContext(ctx, a.runner.gotestsumExecutable, args...)
-	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
+	var output strings.Builder
+	cmd.Stdout = io.MultiWriter(os.Stdout, &output)
+	cmd.Stderr = io.MultiWriter(os.Stderr, &output) // we only need one output
```

Comment:
Needs to buffer everything in memory, which seems fine. A stream-based approach would be more efficient but overkill.